crates/datasources/src/excel/errors.rs

```diff
@@ -0,0 +1,26 @@
+use calamine;
+use datafusion::arrow::error::ArrowError;
+use datafusion::error::DataFusionError;
+use datafusion_ext::errors::ExtensionError;
+
+#[derive(Debug, thiserror::Error)]
+pub enum ExcelError {
+    #[error("Failed to load XLSX: {msg}")]
+    Load { msg: String },
+    #[error("Failed to create record batch: {0}")]
+    CreateRecordBatch(#[from] ArrowError),
+    #[error("Failed to open workbook: {0}")]
+    OpenWorkbook(#[from] calamine::XlsxError),
```

Comment:
Have generic `CalamineError` and `ArrowError`, with specifics for these "create record batch and Open Workbook cases"